### PR TITLE
Use new class AxisAlignedBoxShapeComponentMode to have the correct display name in subedit

### PR DESCRIPTION
**Pull request by guillaume-haerinck**
Signed-off-by: guillaume-haerinck <guillaume.haerinck@outlook.com>

## What does this PR do?

Fix https://github.com/o3de/o3de/issues/10937

Created a new class which inherit the base behavior, the only override is for the name

## How was this PR tested?

Added an axis-aligned box component in an entity and enter subedit

### CODE REVIEW

File: Gems/LmbrCentral/Code/Source/Shape/EditorAxisAlignedBoxShapeComponentMode.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    class EditorAxisAlignedBoxShapeComponentMode : public AzToolsFramework::BoxComponentMode
+    {
+    public:
+        AZ_CLASS_ALLOCATOR_DECL
+
+        EditorAxisAlignedBoxShapeComponentMode(const AZ::EntityComponentIdPair& entityComponentIdPair, AZ::Uuid componentType);
+
+        AZStd::string GetComponentModeName() const override;
+    };
+}
```
Comment (hultonha) [inline Gems/LmbrCentral/Code/Source/Shape/EditorAxisAlignedBoxShapeComponentMode.h:24]: uber nit: Closing namespace name

File: Gems/LmbrCentral/Code/Source/Shape/EditorAxisAlignedBoxShapeComponentMode.h [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+namespace LmbrCentral
+{
+    class EditorAxisAlignedBoxShapeComponentMode : public AzToolsFramework::BoxComponentMode
+    {
+    public:
+        AZ_CLASS_ALLOCATOR_DECL
+
+        EditorAxisAlignedBoxShapeComponentMode(const AZ::EntityComponentIdPair& entityComponentIdPair, AZ::Uuid componentType);
+
+        AZStd::string GetComponentModeName() const override;
```
Comment (hultonha) [inline Gems/LmbrCentral/Code/Source/Shape/EditorAxisAlignedBoxShapeComponentMode.h:22]: suggestion: We quite often include a comment for the overrides to group them together, so here we'd have...

```c++
// AzToolsFramework::BoxComponentMode overrides ...
AZStd::string GetComponentModeName() const override;
```

**Review by hultonha — APPROVED** [review]
Great spot and nice fix! 😄 My comments are super minor but otherwise looks great 👍

File: Gems/LmbrCentral/Code/Source/Shape/EditorAxisAlignedBoxShapeComponentMode.h (hunk 2 above)
Comment (guillaume-haerinck) [reply Gems/LmbrCentral/Code/Source/Shape/EditorAxisAlignedBoxShapeComponentMode.h:23]: Done :) Thanks for the infos !

Comment (hultonha) [reply Gems/LmbrCentral/Code/Source/Shape/EditorAxisAlignedBoxShapeComponentMode.h:23]: No problem! I'll kick off a build now 👍

**Review by greerdv — APPROVED** [review]

**Review by cgalvan — APPROVED** [review]
@guillaume-haerinck Nice you already caught the addition of the new files to the cmake files :) I've kicked off a new AR build

**Pull request merged**